### PR TITLE
Fix issue with Nosara aliasing

### PR DESCRIPTION
**Pull request by withinboredom**
This overwrites the browser's `tk_ai` (anonymous id) cookie with the one stored in user meta. This keeps the id synchronized across all browsers and devices.

#### Changes proposed in this Pull Request:

* Sets the cookie `tk_ai` to the correct parameters to be read from the tracks client
* Updates the user's cookie if it is incorrect
* Calls `jetpack_tracks_get_identity( get_current_user_id() );` asap in plugin initialization in the admin to set the cookie as appropriate, before the js lib can.

#### Testing instructions:

* Login with a user that is not connected
* Navigate to the Jetpack wp-admin with a fresh browser session and check your cookies.
* You should see only one `tk_ai` cookie. If you see more, that's ok. Look for the one with the `/` path, that's your guy.
* Delete your `tk_ai` cookies to simulate using a different/new device and navigate around wp-admin.
* You should have only one `tk_ai` cookie with the previous value.

You can also get creative and set the cookie to a bogus value, it should reset back to the value as before.

### CODE REVIEW

File: class.jetpack.php [hunk 1]
```diff
@@ -3181,6 +3181,8 @@ function admin_init() {
 			add_action( 'jetpack_notices', array( $this, 'alert_auto_ssl_fail' ) );
 		}
 
+		jetpack_tracks_get_identity( get_current_user_id() );
```
Comment (dereksmart) [inline class.jetpack.php]: This feels like a better home, so we're at least making sure that it's only being set if Jetpack is connected.  

https://github.com/Automattic/jetpack/blob/8440afcf9597d5396e40a4b7ebbb330a9d2f349a/class.jetpack-tracks.php#L15

**Comment by withinboredom** [comment]
Huh, weird ... I'm seeing that behavior now too.

**Comment by withinboredom** [comment]
@dereksmart care to give it another swing?

**Comment by withinboredom** [comment]
@dereksmart updated to only check when it's not an ajax request and the user isn't connected.

**Comment by dereksmart** [comment]
@withinboredom beautiful! :shipit:

File: _inc/lib/tracks/class.tracks-client.php [hunk 2]
```diff
@@ -158,7 +158,7 @@ static function get_anon_id() {
 				$anon_id = 'jetpack:' . base64_encode( $binary );
 
 				if ( ! headers_sent() ) {
-					setcookie( 'tk_ai', $anon_id );
+					setcookie( 'tk_ai', $anon_id, time() + 1000, '/' );
```
Comment (gravityrail) [inline _inc/lib/tracks/class.tracks-client.php]: Won't this mean that the cookie expires about 15 minutes after it's set? This seems like it would cause some disruption. Maybe we could check if the value is correct or not and use that to update the cookie, instead of just making the old one expire.

File: _inc/lib/tracks/client.php [hunk 3]
```diff
@@ -86,15 +86,23 @@ function jetpack_tracks_get_identity( $user_id ) {
 		);
 	}
 
+	if ( 0 === $user_id ) {
```
Comment (gravityrail) [inline _inc/lib/tracks/client.php]: How is this change related to the task in question? It does seem like it could be a bug, of course... maybe we should write a test to cover this?

File: class.jetpack-tracks.php [hunk 4]
```diff
@@ -13,6 +13,10 @@ static function track_jetpack_usage() {
 			return;
 		}
 
+		if ( ! defined( 'DOING_AJAX' ) && ! Jetpack::is_user_connected( $user_id = get_current_user_id() ) ) {
```
Comment (gravityrail) [inline class.jetpack-tracks.php]: Can you explain why this change is necessary?

Comment (withinboredom) [reply class.jetpack-tracks.php]: See https://github.com/Automattic/jetpack/pull/7620#discussion_r135115869

File: _inc/lib/tracks/client.php (hunk 3 above)
Comment (withinboredom) [reply _inc/lib/tracks/client.php]: For the same reason we excluded ajax requests (user id 0 for logged out user). We don't want to cookie logged out user.

File: _inc/lib/tracks/class.tracks-client.php (hunk 2 above)
Comment (withinboredom) [reply _inc/lib/tracks/class.tracks-client.php]: How would it cause a disruption?

**Comment by withinboredom** [comment]
Closing because this actually fixes the issue but exposed a worse bug.
